src/observer/index.js

```diff
@@ -189,11 +206,11 @@ function defineReactive (obj, key, val) {
           }
         }
       }
-      return val
+      return target.val
```

Comment:
Not sure how I missed replacing this one out; I'll add a test condition which enforces only calling the underlying getter once.